src/algebra/algebra/basic.lean

```diff
@@ -321,6 +322,22 @@ end ring
 
 end algebra
 
+namespace opposite
+
+variables {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
+
+instance : algebra R Aᵒᵖ :=
+{ smul := λ c x, op (c • unop x),
+  to_ring_hom := (algebra_map R A).to_opposite $ λ x y, algebra.commutes _ _,
+  smul_def' := λ c x, by dsimp; simp only [op_mul, algebra.smul_def, algebra.commutes, op_unop],
+  commutes' := λ r, op_induction $ λ x, by dsimp; simp only [← op_mul, algebra.commutes] }
+
+@[simp] lemma smul_op (c : R) (a : A) : c • op a = op (c • a) := rfl
```

Comment:
This simp-lemma works in the opposite direction of things like `f (a * b) = f a * f b`. Is that intentional?
The simp-lemma for `op_sum` (below) is stated in the "usual" direction.